presto-main/src/test/java/com/facebook/presto/type/BenchmarkBigIntOperators.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.LongBinaryOperator;
+import java.util.function.LongUnaryOperator;
+
+@State(Scope.Thread)
+@Fork(2)
+@Warmup(iterations = 100, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 100, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.Throughput)
+public class BenchmarkBigIntOperators
+{
+    private long leftOperand0;
+    private long leftOperand1;
+    private long leftOperand2;
+    private long leftOperand3;
+    private long leftOperand4;
+
+    private long rightOperand0;
+    private long rightOperand1;
+    private long rightOperand2;
+    private long rightOperand3;
+    private long rightOperand4;
+
+    @Setup
+    public void setup()
+    {
+        leftOperand0 = 1;
+        leftOperand1 = 20;
+        leftOperand2 = 33;
+        leftOperand3 = 407;
+        leftOperand4 = 7890;
+
+        rightOperand0 = 123456;
+        rightOperand1 = 9003;
+        rightOperand2 = 809;
+        rightOperand3 = 67;
+        rightOperand4 = 5;
+    }
+
+    @Benchmark
+    public Object overflowChecksAdd()
+    {
+        return execute(BigintOperators::add);
```

Comment:
@cberner: here are results for this run. I'm sharing to save you ~30 mins of benchmark run:
```
# Run complete. Total time: 00:33:28

Benchmark                                         Mode  Cnt          Score         Error  Units
BenchmarkBigIntOperators.baseLineAdd             thrpt  200   82927195.168 ± 1084007.019  ops/s
BenchmarkBigIntOperators.baseLineDivide          thrpt  200    4439147.805 ±   34827.014  ops/s
BenchmarkBigIntOperators.baseLineMultiply        thrpt  200   76975057.209 ±   68077.203  ops/s
BenchmarkBigIntOperators.baseLineNegate          thrpt  200  154945732.278 ± 1293694.987  ops/s
BenchmarkBigIntOperators.baseLineSubtract        thrpt  200   74956468.886 ± 1471636.767  ops/s
BenchmarkBigIntOperators.overflowChecksAdd       thrpt  200   56960605.468 ±  337728.659  ops/s
BenchmarkBigIntOperators.overflowChecksDivide    thrpt  200    4297724.463 ±   13351.428  ops/s
BenchmarkBigIntOperators.overflowChecksMultiply  thrpt  200   51537797.247 ±  562097.781  ops/s
BenchmarkBigIntOperators.overflowChecksNegate    thrpt  200  110014053.606 ±  242637.210  ops/s
BenchmarkBigIntOperators.overflowChecksSubtract  thrpt  200   65074229.238 ±  127330.814  ops/s
```
This is bothering me:
1. add slowdown ~30%
2. multiply slowdown ~30%
3. negate slowdown ~30%
4. subtract slowdown ~13%

Divide is only ~3% slower.
Errors: mostly below 1% but never exceeding 2%.